function-aws-api-proxy/src/main/java/io/micronaut/function/aws/proxy/payload2/APIGatewayV2HTTPEventServletRequest.java

```diff
@@ -55,7 +55,11 @@ public APIGatewayV2HTTPEventServletRequest(
         super(
             conversionService,
             requestEvent,
-            URI.create(requestEvent.getRequestContext().getHttp().getPath()),
+            ApiGatewayServletRequest.buildUri(
+                requestEvent.getRequestContext().getHttp().getPath(),
```

Comment:
Why not use `rawPath` and `rawQuery` ?

Reply:
What would be the advantage of using rawPath and rawQuery? 🤔

Reply:
You don't need to parse and treat it